instrumentation/instrumentation.py

```diff
@@ -0,0 +1,85 @@
+from iobeam import iobeam
+
+import datetime
+import time
+
+###########################
+# Template for a basic instrumentation app.
+#
+# This template outlines a script that periodically transmits device instrumentation
+# data (e.g., temperature, humidity, gps, etc.) to iobeam.
+#
+# Instructions:
+#
+# 0. Add your project info below
+#
+# 1. Run with placeholder data from command line:
+#     python instrumentation.py
+#
+# 2. Replace placeholder data with device specific code below in each of the 
+#    template functions
+#
+# 3. Run again with your real data!
+#
+#
+###########################
+
+
+###########################
+# Your project info
+###########################
+PROJECT_ID = # YOUR PROJECT ID HERE (int)
+PROJECT_TOKEN = # YOUR PROJECT TOKEN HERE (String)
+
+###########################
+# Template functions
+# NOTE: Fill your own code here
+###########################
+def get_temperature():
+  return 72 # placeholder
+
+def get_humidity():
+  return 0.6 # placeholder
+
+def get_gps():
+  return [42.359155,-71.0952463] # placeholder
+
+###########################
+# Data collection / transmission
+###########################
+def main():
+  # Build iobeam client
+  # Note: In this example, we are choosing to register this device as a new device, 
+  # and save the device_id to disk (saved in the same directory under `iobeam_device_id`)
+  # (To register as a new device, simply delete that file.)
+  builder = iobeam.ClientBuilder(PROJECT_ID, PROJECT_TOKEN).saveToDisk().registerDevice()
+  client = builder.build()
+
+  # Create data store with schema for transmitted data
+  # Note: Update this schema with any new instrumentation data types
+  measurements = client.createDataStore(["temperature", "humidity", "lat", "long"])
+
+  # Loop
+  while True:
+    # Collect data
+    now = int(time.time()*1000)
+    temperature = get_temperature()
+    humidity = get_humidity()
+    gps = get_gps()
+    gps_lat = gps[0]
+    gps_long = gps[1]
+
+    # Add data points to data store
+    measurements.add(now, {"temperature": temperature, "humidity": humidity, "lat": gps_lat, "long": gps_long})
+
+    # Send data to iobeam
+    client.send()
+
+    # Print out data for debugging
+    dt = datetime.datetime.fromtimestamp(now/1000.0)
+    print 'Sent: (Temperature: %.2f, Humidity: %.2f, (Lat, Long): (%.6f, %.6f) at %s' % (temperature, humidity*100, gps_lat, gps_long, dt)
```

Comment:
I prefer using `print()` as a function and using `format()` instead of `%` because its compatible with both python2 and 3. up to you though:

```python
dt = str(datetime.datetime.formtimestamp(now/1000.0))
print('Sent: (Temperature: {:.2f}, Humidity: {:.2f}, (Lat, Long): ({:.6f}, {:.6f}) at {}'.format(temperature, humidity*100, gps_lat, gps_long, dt))
```